Code/Editor/MainWindow.cpp

```diff
@@ -731,32 +732,84 @@ void MainWindow::InitActions()
+                    &AzToolsFramework::EditorTransformComponentSelectionRequests::GetTransformMode);
+
+                action->setChecked(mode == AzToolsFramework::EditorTransformComponentSelectionRequests::Mode::Translation);
+            })
+        .Connect(
+            &QAction::triggered,
+            []()
+            {
+                EditorTransformComponentSelectionRequestBus::Event(
+                    GetEntityContextId(), &EditorTransformComponentSelectionRequests::SetTransformMode,
+                    EditorTransformComponentSelectionRequests::Mode::Translation);
+            });                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
+    am->AddAction(AzToolsFramework::EditModeRotate, tr("Rotate"))
         .SetIcon(Style::icon("Translate"))
         .SetApplyHoverEffect()
         .SetShortcut(tr("2"))
         .SetToolTip(tr("Rotate (2)"))
         .SetCheckable(true)
         .SetStatusTip(tr("Select and rotate selected object(s)"))
-        .RegisterUpdateCallback(cryEdit, &CCryEditApp::OnUpdateEditmodeRotate);
-    am->AddAction(ID_EDITMODE_SCALE, tr("Scale"))
+        .RegisterUpdateCallback([](QAction* action)
+            {
+                Q_ASSERT(action->isCheckable());
+
+                AzToolsFramework::EditorTransformComponentSelectionRequests::Mode mode;
+                AzToolsFramework::EditorTransformComponentSelectionRequestBus::EventResult(
+                    mode, AzToolsFramework::GetEntityContextId(),
+                    &AzToolsFramework::EditorTransformComponentSelectionRequests::GetTransformMode);
+
+                action->setChecked(mode == AzToolsFramework::EditorTransformComponentSelectionRequests::Mode::Rotation);
+            })
+        .Connect(
+            &QAction::triggered,
+            []()
+            {
+                EditorTransformComponentSelectionRequestBus::Event(
+                    GetEntityContextId(), &EditorTransformComponentSelectionRequests::SetTransformMode,
+                    EditorTransformComponentSelectionRequests::Mode::Rotation);
+            });
+    am->AddAction(AzToolsFramework::EditModeScale, tr("Scale"))
         .SetIcon(Style::icon("Scale"))
         .SetApplyHoverEffect()
         .SetShortcut(tr("3"))
         .SetToolTip(tr("Scale (3)"))
         .SetCheckable(true)
         .SetStatusTip(tr("Select and scale selected object(s)"))
-        .RegisterUpdateCallback(cryEdit, &CCryEditApp::OnUpdateEditmodeScale);
+        .RegisterUpdateCallback([](QAction* action)
```

Comment:
As part of this chance can you also delete the member functions on `CCryEditApp` that are no longer used (e.g. `&CCryEditApp::OnUpdateEditmodeScale` etc..

Reply:
Done